### PR TITLE
Add CXT Studio

### PR DESCRIPTION
**Pull request by ColinKinloch**
Add CXT Studio macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] ~My change requires a change to the documentation.~
- [ ] ~I have updated the documentation accordingly.~
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cxt_studio/info.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            { "pin_a": "B2", "pin_b": "B3" },
+            { "pin_a": "F5", "pin_b": "F6" },
+            { "pin_a": "E6", "pin_b": "F0" }
+        ]
+    },
+    "processor": "atmega32u4",
+    "rgblight": {
+        "pin": "F7",
+        "led_count": 12
+    },
```
Comment (waffle87) [inline keyboards/cxt_studio/info.json:33]:
```suggestion
```

File: keyboards/cxt_studio/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Colin Kinloch (@ColinKinloch)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define RGB_MATRIX_LED_COUNT RGBLED_NUM
```
Comment (waffle87) [inline keyboards/cxt_studio/config.h:6]:
```suggestion
#define RGB_MATRIX_LED_COUNT 12
#define RGB_DI_PIN F7
```

File: keyboards/cxt_studio/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# cxt_studio
+
+![cxt_studio](https://gist.githubusercontent.com/ColinKinloch/cb49a8b42fcd07445232a26f562f3b9d/raw/ff9ac62884d724811f634c07e44c737a148181cc/cxt_studio.jpg)
```
Comment (waffle87) [inline keyboards/cxt_studio/readme.md:3]: This image should be rehosted on Imgur.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/cxt_studio/readme.md (hunk 3 above)
Comment (ColinKinloch) [reply keyboards/cxt_studio/readme.md:3]: I can't seem to create an account. If the issue is the risk of me modifying the image then I could change it to "https://user-images.githubusercontent.com/334272/211168668-84c1cc8c-e5c5-45e7-a703-2fd47007131e.jpg" which is generated from embedding the image in this comment.
Screenshot 2023-01-07 at 20-08-16 Imgur
cxt_studio

Comment (ColinKinloch) [reply keyboards/cxt_studio/readme.md:3]: Or you could upload it?

Comment (waffle87) [reply keyboards/cxt_studio/readme.md:3]: Yes:
`cxt_studio`

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/cxt_studio/cxt_studio.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Colin Kinloch (@ColinKinloch)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+static uint8_t anim = 0;
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/cxt_studio/cxt_studio.c:8]:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
```

File: keyboards/cxt_studio/cxt_studio.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		anim = 0;
+	} else if (anim < 0) {
+		anim = RGB_MATRIX_EFFECT_MAX - 1;
+	}
+	rgblight_mode(anim);
+      }
+      break;
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/cxt_studio/cxt_studio.c:53]:
```suggestion
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/cxt_studio/keymaps/default/keymap.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard": "cxt_studio",
+    "keymap": "default",
+    "layout": "LAYOUT",
+    "layers": [
+        [
+            "KC_ESC",     "KC_F11",              "KC_NO",               "KC_MEDIA_STOP",
+            "KC_NO",      "KC_NO",               "KC_MEDIA_REWIND",     "KC_MEDIA_FAST_FORWARD",
+            "KC_NO",      "KC_MEDIA_PLAY_PAUSE", "KC_MEDIA_PREV_TRACK", "KC_MEDIA_NEXT_TRACK",
+            "KC_KB_MUTE", "KC_NO",               "KC_NO",               "RGB_TOG"
```
Comment (fauxpark) [inline keyboards/cxt_studio/keymaps/default/keymap.json:10]:
```suggestion
            "KC_ESC",  "KC_F11",  "KC_NO",   "KC_MSTP",
            "KC_NO",   "KC_NO",   "KC_MRWD", "KC_MFFD",
            "KC_NO",   "KC_MPLY", "KC_MPLY", "KC_MNXT",
            "KC_MUTE", "KC_NO",   "KC_NO",   "RGB_TOG"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by fauxpark** [comment]
That is probably some kind of counterfeit MCU. It likely does not even run QMK.

**Comment by ColinKinloch** [comment]
@fauxpark plugging it in with top left button held inters into DFU mode. 
The chip on mine is a MEGA32U4
IMG_20231218_161121

**Comment by ColinKinloch** [comment]
I bought mine from "[eParts Store](https://www.aliexpress.com/store/911767268)" on aliexpress in December 2022 https://www.aliexpress.com/item/1005002462322657.html

**Comment by deepu105** [comment]
Mine definitely is running QMK as I can configure it using standard via and
vial apps. Just couldn't get into DFU

**Comment by fauxpark** [comment]
Just because it can be configured using VIA doesn't necessarily mean it's running QMK. The protocol and code are open source; we suspect some board makers have reimplemented it for their own firmware.

**Comment by sigprof** [comment]
> Mine has a chip mark3ed "Fpapss USB G3G6"

@deepu105 Can you post some photos of the back side of the PCB?  This would probably be the only way to determine whether you actually have some different device that was sold under the same (or a very similar) name as the supported one.

**Comment by Igorkoza** [comment]
Hey guys, I believe I have the same board as @deepu105 
I bought it last month from AliExpress, and I've tried everything possible to enter the bootloader, but nothing worked....
This one works with VIA with no problems, and If I press the back reset button and plug the USB, all the lights stay off for about 5 seconds, then they light, but still the QMK MSYS does not recognize it.

Here are some pics
Imagem do WhatsApp de 2024-03-19 à(s) 23 28 52_85c4e351

Capturar

I'm a complete noob on programming and my only problem with this board is that it lacks a leyer indicator led, so if It supports QMK I'll only use it to set per layer backlight color, but maybe someone with more experience can give us some ideas to make it work.

**Comment by sigprof** [comment]
@Igorkoza 
Unfortunately, this looks like a completely different board with some unknown MCU that is not supported by QMK.

**Pull request merged**